src/primitives/orchard.h

```diff
@@ -52,87 +52,88 @@ class OrchardBundle
     OrchardBundle& operator=(const OrchardBundle& bundle)
     {
         if (this != &bundle) {
-            inner.reset(orchard_bundle_clone(bundle.inner.get()));
+            inner = bundle.inner->box_clone();
         }
         return *this;
     }
 
-    rust::Box<orchard_bundle::Bundle> GetDetails() const {
-        return orchard_bundle::from_tx_bundle(reinterpret_cast<orchard_bundle::OrchardBundle*>(inner.get()));
+    const rust::Box<orchard_bundle::Bundle>& GetDetails() const {
```

Comment:
I checked all uses of this to ensure that the changed return type would not break any assumptions of the calling code.